src/features/authentication/components/LogisterScreen.tsx

```diff
@@ -31,75 +33,137 @@ const LogisterScreen = ({ }: LogisterScreenProps): JSX.Element => {
     const [confirmPasswordError, setConfirmPasswordError] = useState('')
 
     const onSignup = () => {
-        if (emailError || passwordError || confirmPasswordError) {
+        if (emailError || passwordError || confirmPasswordError || !email || !password || !confirmPassword) {
             Alert.alert(
                 'Oh oh!',
                 'Please fill out the form correctly.',
                 [{ text: 'Okay', style: 'destructive' }]
             )
         }
-        else console.log('Signup')
+        else {
+            auth()
+                .createUserWithEmailAndPassword(email, password)
+                .then(() => {
+                    console.log('User account created & signed in!')
+                })
+                .catch((error: FirebaseAuthTypes.NativeFirebaseAuthError) => {
+                    errorHandler(error)
+                })
+        }
+    }
+
+    const onLogin = () => {
+        if (emailError || passwordError || confirmPasswordError || !email || !password) {
+            Alert.alert(
+                'Oh oh!',
+                'Please fill out the form correctly.',
+                [{ text: 'Okay', style: 'destructive' }]
+            )
+        }
+        else {
+            auth().signInWithEmailAndPassword(email, password)
+                .then(() => {
+                    console.log('User account signed in!')
+                })
+                .catch((error: FirebaseAuthTypes.NativeFirebaseAuthError) => {
+                    errorHandler(error)
+                })
+        }
+    }
+
+    const errorHandler = (error: FirebaseAuthTypes.NativeFirebaseAuthError) => {
+        let msg = 'oops! something went wrong'
+
+        if (error.code === 'auth/email-already-in-use')
```

Comment:
add the errors to strings file

Reply:
I just moved the whole errorHandler to its own file, is it still necessary to move the errors to strings file?